source/Ship.cpp

```diff
@@ -2257,6 +2257,26 @@ bool Ship::IsOverheated() const
 
 
 
+bool Ship::IsIonized() const
+{
+	if(!ionization)
+		return false;
+
+	// A ship can only be fully ionized if its engines or weapons require energy.
+	bool movementEnergy = attributes.Get("thrusting energy") > 0
+		|| attributes.Get("reverse thrusting energy") > 0
+		|| attributes.Get("turning energy") > 0;
+
+	bool firingEnergy = false;
+	for(const auto &it : outfits)
+		if(it.first->IsWeapon() && it.first->FiringEnergy() > 0)
+			firingEnergy = true;
```

Comment:
We don't need to keep looping if we've found one weapon that requires energy.
```suggestion
	for(const auto &it : outfits)
		if(it.first->IsWeapon() && it.first->FiringEnergy() > 0)
		{
			firingEnergy = true;
			break;
		}
```